packages/airnode-node/src/evm/templates/template-fetching.test.ts

```diff
@@ -15,9 +15,61 @@ describe('fetch (templates)', () => {
     mutableFetchOptions = {
       airnodeRrpAddress: '0xD5659F26A72A8D718d1955C42B3AE418edB001e0',
       provider: new ethers.providers.JsonRpcProvider(),
+      configTemplates: [],
+      airnodeAddress: '0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266',
     };
   });
 
+  it('uses valid templates from config without fetching from the chain', async () => {
+    mutableFetchOptions = {
+      ...mutableFetchOptions,
+      configTemplates: [
+        {
+          templateId: '0x38ba0e80224f14d0c654c4ba6e3745fcb7f310fd4f2f80994fe802da013edafe',
+          endpointId: '0x13dea3311fe0d6b84f4daeab831befbc49e19e6494c41e9e065a09c3c68f43b6',
+          encodedParameters: '0x6874656d706c6174656576616c7565',
+        },
+      ],
+    };
+
+    const rawTemplates = {
+      airnodes: ['airnode-0', '0xD5659F26A72A8D718d1955C42B3AE418edB001e0'],
+      endpointIds: ['endpointId-0', '0x13dea3311fe0d6b84f4daeab831befbc49e19e6494c41e9e065a09c3c68f43b6'],
+      parameters: ['0x6874656d706c6174656576616c7565', '0x6874656d706c6174656576616c7565'],
+    };
+    getTemplatesMock.mockResolvedValueOnce(rawTemplates);
+
+    const apiCall1 = fixtures.requests.buildApiCall({
+      templateId: '0x38ba0e80224f14d0c654c4ba6e3745fcb7f310fd4f2f80994fe802da013edafe',
+    });
+    const apiCall2 = fixtures.requests.buildApiCall({ templateId: 'templateId-0' });
+
+    const [logs, res] = await templates.fetch([apiCall1, apiCall2], mutableFetchOptions);
+    expect(logs).toEqual([]);
+    expect(res).toEqual({
+      'templateId-0': {
+        airnodeAddress: 'airnode-0',
+        endpointId: 'endpointId-0',
+        encodedParameters: '0x6874656d706c6174656576616c7565',
+        id: 'templateId-0',
+      },
+      '0x38ba0e80224f14d0c654c4ba6e3745fcb7f310fd4f2f80994fe802da013edafe': {
+        airnodeAddress: '0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266',
+        endpointId: '0x13dea3311fe0d6b84f4daeab831befbc49e19e6494c41e9e065a09c3c68f43b6',
+        encodedParameters: '0x6874656d706c6174656576616c7565',
+        id: '0x38ba0e80224f14d0c654c4ba6e3745fcb7f310fd4f2f80994fe802da013edafe',
+      },
+    });
+
+    // Not called with a valid config templateId
+    expect(templatesMock).not.toHaveBeenCalled();
```

Comment:
Are these assertions correct? I'd expect:
```
expect(getTemplatesMock).not.toHaveBeenCalledWith('0x38ba0e80224f14d0c654c4ba6e3745fcb7f310fd4f2f80994fe802da013edafe');
```

and
 
```
    expect(getTemplatesMock).toHaveBeenCalledWith([
      'templateId-0',
    ]);
```

Reply:
I believe they are correct because the `fetchTemplateGroup` function first calls the `getTemplates` contract method:

```ts
const fetchTemplateGroup = () => 
{...
const contractCall = () => airnodeRrp.getTemplates(templateIds);
  const [err, rawTemplates] = await go(contractCall, { retries: 1, timeoutMs: DEFAULT_RETRY_TIMEOUT_MS });
  // If we fail to fetch templates, the linked requests will be discarded and retried
  // on the next run
  if (err || !rawTemplates) {
    const groupLog = logger.pend('ERROR', 'Failed to fetch API call templates', err);

    // If the template group cannot be fetched, fallback to fetching templates individually
    const promises = templateIds.map((id) => fetchTemplate(airnodeRrp, id));
....
```

In the test we **mockResolvedValue** for `getTemplates`:

```ts
const rawTemplates = {
      airnodes: ['airnode-0', '0xD5659F26A72A8D718d1955C42B3AE418edB001e0'],
      endpointIds: ['endpointId-0', '0x13dea3311fe0d6b84f4daeab831befbc49e19e6494c41e9e065a09c3c68f43b6'],
      parameters: ['0x6874656d706c6174656576616c7565', '0x6874656d706c6174656576616c7565'],
    };
    getTemplatesMock.mockResolvedValueOnce(rawTemplates);
```

So as I understand, it should not be fetching templates individually.

Reply:
I think the comments mostly confused me, but I think this looks better now.